File: scrape.py
import urllib.request
import re

url = 'https://www.apple.com/kr/'
req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
try:
    html = urllib.request.urlopen(req).read().decode('utf-8')
    
    # Extract h2 and h3
    print("--- Titles ---")
    titles = re.findall(r'<h[23][^>]*>(.*?)</h[23]>', html, re.DOTALL | re.IGNORECASE)
    for t in titles[:20]:
        clean = re.sub(r'<[^>]+>', '', t).strip()
        clean = re.sub(r'\s+', ' ', clean)
        print(clean)
        
    print("--- Image URLs ---")
    images = re.findall(r'(https://[^\s\"\'\)]+\.(?:jpg|png|jpeg))', html, re.IGNORECASE)
    seen = set()
    for img in images:
        if img not in seen:
            print(img)
            seen.add(img)
except Exception as e:
    print(e)

File: scrape.js
const https = require('https');

https.get('https://www.apple.com/kr/', (resp) => {
  let data = '';

  resp.on('data', (chunk) => {
    data += chunk;
  });

  resp.on('end', () => {
    const titleRegex = /<h[23][^>]*>(.*?)<\/h[23]>/gi;
    const imgRegex = /(https:\/\/[^\s\"\'\)]+\.(?:jpg|png|jpeg))/gi;
    
    console.log("--- Titles ---");
    let titleMatch;
    let titleCount = 0;
    while ((titleMatch = titleRegex.exec(data)) !== null && titleCount < 20) {
      let clean = titleMatch[1].replace(/<[^>]+>/g, '').trim().replace(/\s+/g, ' ');
      console.log(clean);
      titleCount++;
    }

    console.log("--- Image URLs ---");
    let imgMatch;
    let seen = new Set();
    while ((imgMatch = imgRegex.exec(data)) !== null) {
      if (!seen.has(imgMatch[1])) {
        console.log(imgMatch[1]);
        seen.add(imgMatch[1]);
      }
    }
  });
}).on("error", (err) => {
  console.log("Error: " + err.message);
});
